template.tpl

```diff
@@ -897,18 +897,52 @@ const isSpContextsProp = (prop) => {
   return prop.indexOf('x-sp-contexts_') === 0;
 };
 
+/**
+ * Removes the major version part from a schema reference if exists.
+ * @example
+ * // returns 'com_acme_test'
+ * mkVersionFree('com_acme_test_1')
+ * @example
+ * // returns 'com_acme_test'
+ * mkVersionFree('com_acme_test')
+ *
+ * @param {string} schemaRef - The schema
+ * @returns {string}
+ */
+const mkVersionFree = (schemaRef) => {
+  const versionRexp = createRegex('_[0-9]+$', 'g');
+  return schemaRef.replace(versionRexp, '');
+};
+
 /**
  * Given a list of entity references and an entity name,
  * returns the index of a matching reference.
  * Matching reference means whether the entity name starts with ref.
+ * @example
+ * // returns 0
+ * getReferenceIdx('com_test_test_1', ['com_test_test_1']);
+ * @example
+ * // returns 0
+ * getReferenceIdx('com_test_test_1', ['com_test_test']);
+ * @example
+ * // returns -1
+ * getReferenceIdx('com_test_test_1', ['com_test_test_2']);
+ * @example
+ * // returns -1
+ * getReferenceIdx('com_test_test', ['com_test_test_fail']);
+ * @example
+ * // returns -1
+ * getReferenceIdx('com_test_test_fail', ['com_test_test']);
  *
  * @param {string} entity - The entity name to match
- * @param {string[]} refsList - An array of strings
+ * @param {string[]} refsList - An array of references
  * @returns {integer}
  */
 const getReferenceIdx = (entity, refsList) => {
   for (let i = 0; i < refsList.length; i++) {
-    if (entity.indexOf(refsList[i]) === 0) {
+    const okControl = entity.indexOf(refsList[i]) === 0;
+    const okFree = mkVersionFree(entity) === mkVersionFree(refsList[i]);
```

Comment:
Hey @jethron , thanks for reviewing!

> Is there supposed to be a case where okControl && !okFree

Yes, this is exactly the bug we are solving for. As one of the examples in the description, this would be the case for

```
getReferenceIdx('com_test_test_fail', ['com_test_test'])
// or even
getReferenceIdx('com_test_test_fail_1', ['com_test_test'])
```

> I think this is missing a test of whether the compared values are free/control,

Can you please provide an example to test?

> (e.g. free-include a mapping for youtube, control-exclude youtube v1, payload with youtube v2 is still excluded but shouldn't be?)

Payload with v2 will not be excluded. You can find such test [included here](https://github.com/snowplow/snowplow-gtm-server-side-braze-tag/blob/1abab639ade6055d526df69c576b3ce675ab7238/template.tpl#L2465), where the media_player entity is [free included](https://github.com/snowplow/snowplow-gtm-server-side-braze-tag/blob/1abab639ade6055d526df69c576b3ce675ab7238/template.tpl#L2489-L2494) and its [v2 is control-excluded](https://github.com/snowplow/snowplow-gtm-server-side-braze-tag/blob/1abab639ade6055d526df69c576b3ce675ab7238/template.tpl#L2520-L2523). The result is that v1 data is still [included in the payload](https://github.com/snowplow/snowplow-gtm-server-side-braze-tag/blob/1abab639ade6055d526df69c576b3ce675ab7238/template.tpl#L2547-L2556).

In general, the expected behavior is:

|         |               |               |               |      |
|---------|---------------|---------------|---------------|------|
| Include | Control on v1 | Control on v1 | Free          | Free |
| Exclude | Control on v1 | Free          | Control on v1 | Free |
| v1      | -             | -             | **-**             | -    |
| v2      | +             | -             | +             | -    |

What do you think?

Reply:
All good, think I confused myself -- totally my mistake. Thanks for clarifying! :)